roles/ceph-client/tasks/create_users_keys.yml

```diff
@@ -1,94 +1,83 @@
 ---
-- name: set docker_exec_client_cmd_binary to ceph-authtool
+- name: set_fact delegated_node
   set_fact:
-    docker_exec_client_cmd_binary: ceph-authtool
-  when: containerized_deployment
+    delegated_node: "{{ groups[mon_group_name][0] if groups.get(mon_group_name, []) | length > 0 else inventory_hostname }}"
 
-- name: set docker_exec_client_cmd for containers
+- name: set_fact condition_copy_admin_key
   set_fact:
-    docker_exec_client_cmd: docker run --rm -v /etc/ceph:/etc/ceph --entrypoint /usr/bin/{{ docker_exec_client_cmd_binary }} {{ ceph_docker_registry}}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }}
-  when: containerized_deployment
+    condition_copy_admin_key: "{{ True if groups.get(mon_group_name, []) | length > 0 else copy_admin_key }}"
 
-- name: set docker_exec_client_cmd for non-containers
+- name: set_fact docker_exec_cmd
   set_fact:
-    docker_exec_client_cmd: ceph-authtool
-  when: not containerized_deployment
+    docker_exec_cmd: "docker exec {% if groups.get(mon_group_name, []) | length > 0 -%} ceph-mon-{{ hostvars[delegated_node]['ansible_hostname'] }} {% else %} ceph-client-{{ hostvars[delegated_node]['ansible_hostname'] }} {% endif %}"
```

Comment:
This is wrong, if there is no mon, there is no container running `ceph-client-XXXX`.

Reply:
Actually, this commit cannot be easily backported unless we take a lot of what's in stable-3.1/master...